gnucash/gnome-utils/dialog-properties.c

```diff
@@ -0,0 +1,1702 @@
+
+    LEAVE("added content to page");
+}
+
+
+static gint
+tab_cmp (GtkWidget *page_a, GtkWidget *page_b, GtkNotebook *notebook)
+{
+    return g_utf8_collate (gtk_notebook_get_tab_label_text (notebook, page_a),
+                           gtk_notebook_get_tab_label_text (notebook, page_b));
+}
+
+
+static void
+gnc_props_sort_pages (GtkNotebook *notebook)
+{
+    gint n_pages, i;
+    GList *tabs = NULL, *iter = NULL;
+
+    g_return_if_fail (GTK_IS_NOTEBOOK (notebook));
+
+    /* gather tabs */
+    n_pages = gtk_notebook_get_n_pages (notebook);
+    for (i = n_pages - 1; i >= 0; i--)
+        tabs = g_list_prepend (tabs, gtk_notebook_get_nth_page (notebook, i));
+
+    /* sort in local copy */
+    tabs = g_list_sort_with_data (tabs, (GCompareDataFunc) tab_cmp, notebook);
+
+    /* reorder tabs */
+    for (i = 0, iter = tabs; iter; i++, iter = iter->next)
+        gtk_notebook_reorder_child (notebook, GTK_WIDGET (iter->data), i);
+
+    g_list_free (tabs);
+}
+
+
+static void
+gnc_properties_do_refresh (void)
+{
+    QofBook *book = gnc_get_current_book ();
+    gint use_read_only_threshold_before = qof_book_get_num_days_autoreadonly (book);
+
+    gboolean use_split_action_for_num_after;
+//    gboolean use_book_currency_after = gnc_book_use_book_currency (book);
+    gint use_read_only_threshold_after;
+
+    gboolean do_refresh = FALSE;
+
+    // mark cached value as invalid so we get new value
+    book->cached_num_days_autoreadonly_isvalid = FALSE;
+    use_read_only_threshold_after = qof_book_get_num_days_autoreadonly (book);
+
+    // mark cached value as invalid so we get new value
+    book->cached_num_field_source_isvalid = FALSE;
+    use_split_action_for_num_after = qof_book_use_split_action_for_num_field (book);
+
+    if (use_split_action_for_num_before != use_split_action_for_num_after)
+    {
+        gnc_book_option_num_field_source_change_cb (
+                                                use_split_action_for_num_after);
+        do_refresh = TRUE;
+    }
+
+//    if (use_book_currency_before != use_book_currency_after)
+//    {
+//        gnc_book_option_book_currency_selected_cb (use_book_currency_after);
+//        do_refresh = TRUE;
+//    }
+
+    if (use_read_only_threshold_before != use_read_only_threshold_after)
+        do_refresh = TRUE;
+
+    if (do_refresh)
+        gnc_gui_refresh_all ();
+}
+
+/*******************************/
+/* Dynamically added Callbacks */
+/*******************************/
+
+static void
+gnc_props_split_widget_name (const gchar *name, gchar **path)
+{
+    const gchar *group_with_pref = name + PROP_PREFIX_LEN;
+    *path = g_strdup (group_with_pref);
+}
+
+/****************************************************************************/
+
+/** Connect a GtkFontButton widget to its properties value.
+ *
+ *  @internal
+ *
+ *  @param fb A pointer to the font button that should be connected.
+ */
+static void
+gnc_props_connect_font_button (GtkFontButton *fb)
+{
+    gchar *path;
```

Comment:
Init to NULL. This (and similar in other functions, `free` at lines 688, 964, 1029, 1049) causes a "might be used uninitialized warning and fails to build. That's the current cause of the test failures.

Reply:
Sorry about that, missed this as travis is not running on my master, need to look into that.